csharp/ql/lib/ext/System.model.yml

```diff
@@ -374,12 +374,37 @@ extensions:
       - ["System", "Lazy<T>", False, "Lazy", "(System.Func<T>,System.Boolean)", "", "Argument[0].ReturnValue", "Argument[this].Property[System.Lazy`1.Value]", "value", "manual"]
       - ["System", "Lazy<T>", False, "Lazy", "(System.Func<T>,System.Threading.LazyThreadSafetyMode)", "", "Argument[0].ReturnValue", "Argument[this].Property[System.Lazy`1.Value]", "value", "manual"]
       - ["System", "Lazy<T>", False, "get_Value", "()", "", "Argument[this]", "ReturnValue", "taint", "manual"]
+      - ["System", "MemoryExtensions", False, "Replace<T>", "(System.Span<T>,T,T)", "", "Argument[2]", "Argument[0].Element", "value", "manual"]
+      - ["System", "MemoryExtensions", False, "Replace<T>", "(System.ReadOnlySpan<T>,System.Span<T>,T,T)", "", "Argument[0].Element", "Argument[1].Element", "value", "manual"]
```

Comment:
This might introduce false positives, but I don't think we have a more accurate way of modeling it.

Reply:
Yes, it is quite inaccurate. Do you think it is better than we don't include them? It is a pretty big "maybe" that we have value flow.

Reply:
We could check the usages of this method with MRVA, maybe that helps in coming up with a decision.

Reply:
Thinking a bit more about this. We should do the same as we do with `Enumerable.Except`.

Reply:
`Enumerable.Except` just has flow from input to output (not taking the `except` list into account).
The `MemoryExtensions.Replace` method probably resembles `Enumerable.DefaultIfEmpty` a bit more in terms of whether we should include inaccurate summaries.
For `DefaultIsEmpty` we have
```
    - ["System.Linq", "Enumerable", False, "DefaultIfEmpty<TSource>", "(System.Collections.Generic.IEnumerable<TSource>,TSource)", "", "Argument[1]", "ReturnValue", "value", "manual"]
```
which is also quite inaccurate as we only have flow from `Argument[1]` in case the enumerable is empty, but we still include a summary for it; Following that practice we should keep both summaries.

Reply:
Hmm, the `DefaultIsEmpty` summaries are also incorrect in terms of the return. I will update them according to their intent.

Reply:
I just checked `Except`:
```
["System.Linq", "Enumerable", False, "Except<TSource>", "(System.Collections.Generic.IEnumerable<TSource>,System.Collections.Generic.IEnumerable<TSource>)", "", "Argument[0].Element", "ReturnValue", "value", "manual"]
```

shouldn't this also be `"ReturnValue.Element"` instead of `"ReturnValue"`?

Reply:
Coming back to the original question of `System.MemoryExtensions.Replace<T>`. I think we can conclude that the current model is okay.

Reply:
Hah, you are right about `Except`. I have updated the models as data rows for `Except` and `DefaultIfEmpty` in `System.Linq.model.yml` and checked if there are other rows where the target was `ReturnValue`, but I couldn't find other besides the two we just identified.

Reply:
When/if the PR gets approved, I will run DCA again.